### PR TITLE
nixos/prometheus-exporters/py-air-control: init

### PR DESCRIPTION
**Pull request by urbas**
###### Motivation for this change

This change adds the `py-air-control` Prometheus Exporter Module for NixOS.

This module sets up the `prometheus-py-air-control-exporter` service which gathers air quality information from an air quality/air purification device connected to the network.

This is a follow-up to the PR #105966.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`) (tested the execution of this new `prometheus-py-air-control-exporter` service on a Raspberry Pi with an actual air purifier device)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @SuperSandro2000 @lukegb

### CODE REVIEW

File: nixos/modules/services/monitoring/prometheus/exporters/py-air-control.nix [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+  py-air-control-exporter-env = pkgs.python3.withPackages (pyPkgs: [
+      pyPkgs.py-air-control-exporter
+  ]);
+
+  workingDir = "/var/lib/${cfg.stateDir}";
+
+in
+{
+  port = 9896;
```
Comment (urbas) [inline nixos/modules/services/monitoring/prometheus/exporters/py-air-control.nix:16]: This port has been added to the [Prometheus' port allocations table](https://github.com/prometheus/prometheus/wiki/Default-port-allocations).

**Comment by urbas** [comment]
Added tests. They will fail until the `py-air-control-exporter` package is updated to 0.2.0.

File: nixos/modules/services/monitoring/prometheus/exporters/py-air-control.nix [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  extraOpts = {
+    deviceHostname = mkOption {
+      type = types.str;
+      example = ''"192.168.1.123"'';
```
Comment (WilliButz) [inline nixos/modules/services/monitoring/prometheus/exporters/py-air-control.nix:20]: Was the multi-line string intended here? In the test it is just a simple string without double quotes.

**Review by WilliButz — CHANGES_REQUESTED** [review]
LGTM, just one minor remark :+1:

File: nixos/modules/services/monitoring/prometheus/exporters/py-air-control.nix (hunk 2 above)
Comment (urbas) [reply nixos/modules/services/monitoring/prometheus/exporters/py-air-control.nix]: Oh, that's an accident. Converted to `example = "192.168.1.123";`.

**Comment by WilliButz** [comment]
@GrahamcOfBorg test prometheus-exporters.py-air-control

**Review by WilliButz — APPROVED** [review]

**Pull request merged**